be/src/io/fs/azure_obj_storage_client.cpp

```diff
@@ -203,49 +240,59 @@ ObjectStorageResponse AzureObjStorageClient::delete_objects(const ObjectStorageP
     auto end = std::end(objs);
 
     while (begin != end) {
-        auto batch = _client->CreateBatch();
+        auto deleter = AzureBatchDeleter(_client.get(), opts);
         auto chunkEnd = begin;
         std::advance(chunkEnd, std::min(BlobBatchMaxOperations,
                                         static_cast<size_t>(std::distance(begin, end))));
-        for (auto it = begin; it != chunkEnd; ++it) {
-            batch.DeleteBlob(*it);
-        }
+
+        std::ranges::for_each(std::ranges::subrange(begin, chunkEnd),
```

Comment:
`chunkEnd` style is not correct, pls check other occurrences

and, capture only what is used